examples/server/server.cpp

```diff
@@ -3691,37 +3692,18 @@ int main(int argc, char ** argv) {
             return;
         }
 
+        if (type == SERVER_TASK_TYPE_SLOT_SAVE) {
+            GGML_ASSERT(dynamic_cast<server_task_result_slot_save_load*>(result.get()) != nullptr);
+        }
         res_ok(res, result->to_json());
     };
 
-    const auto handle_slots_restore = [&ctx_server, &res_error, &res_ok, &params](const httplib::Request & req, httplib::Response & res, int id_slot) {
-        json request_data = json::parse(req.body);
-        std::string filename = request_data.at("filename");
-        if (!fs_validate_filename(filename)) {
-            res_error(res, format_error_response("Invalid filename", ERROR_TYPE_INVALID_REQUEST));
-            return;
-        }
-        std::string filepath = params.slot_save_path + filename;
-
-        server_task task(SERVER_TASK_TYPE_SLOT_RESTORE);
-        task.id = ctx_server.queue_tasks.get_new_id();
-        task.slot_action.slot_id  = id_slot;
-        task.slot_action.filename = filename;
-        task.slot_action.filepath = filepath;
-
-        ctx_server.queue_results.add_waiting_task_id(task.id);
-        ctx_server.queue_tasks.post(task);
-
-        server_task_result_ptr result = ctx_server.queue_results.recv(task.id);
-        ctx_server.queue_results.remove_waiting_task_id(task.id);
-
-        if (result->is_error()) {
-            res_error(res, result->to_json());
-            return;
-        }
+    const auto handle_slots_save = [&handle_slot_type](const httplib::Request & req, httplib::Response & res, int id_slot) {
+        handle_slot_type(req, res, id_slot, SERVER_TASK_TYPE_SLOT_SAVE);
+    };
 
-        GGML_ASSERT(dynamic_cast<server_task_result_slot_save_load*>(result.get()) != nullptr);
-        res_ok(res, result->to_json());
+    const auto handle_slots_restore = [&handle_slot_type](const httplib::Request & req, httplib::Response & res, int id_slot) {
+        handle_slot_type(req, res, id_slot, SERVER_TASK_TYPE_SLOT_RESTORE);
     };
 
     const auto handle_slots_erase = [&ctx_server, &res_error, &res_ok](const httplib::Request & /* req */, httplib::Response & res, int id_slot) {
```

Comment:
IMO we should also extend the function to be reused by `handle_slots_erase`. The only thing need to adapt is that `SERVER_TASK_TYPE_SLOT_ERASE` ignores `filename`

Reply:
I'll take a look at this too.

Reply:
I've update this in https://github.com/ggerganov/llama.cpp/pull/11535/commits/2d2d07618e8f31b2cb18a0ea2e68869be6d2163c